mixer/adapter/cloudwatch/cloudwatch.go

```diff
@@ -119,6 +132,35 @@ func (b *builder) Validate() (ce *adapter.ConfigErrors) {
 			ce = ce.Append("dimensions", fmt.Errorf("metrics can only contain %v dimensions", dimensionLimit))
 		}
 	}
+
+	// LogGroupName should not be empty
+	if len(b.adpCfg.GetLogGroupName()) == 0 {
```

Comment:
curious: is it possible to use CloudWatch without using the CloudWatch Logs functionality (and vice versa)?

This code seems to imply that that is not possible. Does this change then also break people using outdated config for this adapter?

Reply:
My understanding is that as logentries pushed to CloudwatchLogs are just the metrics generated by mixer but in the form of logs, the user would want to use both. But you’re right that if people use outdated config this change will break the adapter for them. I implemented this in the similar fashion as it is in the stackdriver adapter.
What would you suggest? Should I first check whether the user has given logentry and metric configuration in their operator config file and then perform the validations.

Reply:
My suggestion would be to make them separable. I'm not well-versed in CloudWatch, however.

Having config checks dependent on which configuration was supplied seems reasonable.